src/main/java/uk/wanat/theclinick/controller/PatientController.java

```diff
@@ -0,0 +1,57 @@
+package uk.wanat.theclinick.controller;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import uk.wanat.theclinick.model.Patient;
+import uk.wanat.theclinick.service.PatientServiceImpl;
+
+import java.util.List;
+
+@Controller
+public class PatientController {
+
+    @Autowired
+    PatientServiceImpl patientServiceImpl;
+
+    @RequestMapping("/showAllPatients")
+    public String showPatients(Model model) {
+        model.addAttribute("patientList", patientServiceImpl.findAll());
+        return "patient/patient-view-all";
+    }
+
+    @RequestMapping("/patient/addPatient")
+    public String addPatient(Model model) {
+        model.addAttribute("patient", new Patient());
+        return "patient/add-patient-form";
+}
+
+    @PostMapping("/patient/addPatient")
+    public String addPatient(@ModelAttribute Patient patient, Model model) {
+        List<Patient> all = patientServiceImpl.findAll();
+        for (Patient patientTemp : all) {
+            if (patientTemp.getNationalInsuranceNumber().equals(patient.getNationalInsuranceNumber())) {
+                model.addAttribute("patient", patientTemp);
+                return "patient/patient-already-exists";
+            }
+        }
+        patientServiceImpl.create(patient);
+        return "patient/patient-created";
+    }
+
+    @RequestMapping("patient/{id}/update")
+    public String updateUpdate(@PathVariable String id, Model model) {
```

Comment:
Is this method name what you intended?

Reply:
It was mistakenly written. It should be as below:
1. 
```suggestion
    public String updatePatient(@PathVariable String id, Model model) {
```